javagen/src/main/java/com/azure/autorest/util/MethodUtil.java

```diff
@@ -55,4 +70,155 @@ public static boolean isMethodIncludeRepeatableRequestHeaders(ProxyMethod proxyM
     public static boolean isHttpMethodSupportRepeatableRequestHeaders(HttpMethod httpMethod) {
         return REPEATABILITY_REQUEST_HTTP_METHODS.contains(httpMethod);
     }
+
+    /**
+     * @param requests a list of requests
+     * @return the first request consumes binary type, if no binary request, return the first request in requests
+     *
+     * If the binary request does not have content-type parameter, we will add one for it
+     */
+    public static Request findFirstBinaryRequest(List<Request> requests, Operation operation) {
+        Request selectedRequest = requests.get(0);
+        for (Request request : requests) {
+            if (request.getProtocol().getHttp().getKnownMediaType() != null
+                    && request.getProtocol().getHttp().getKnownMediaType().equals(KnownMediaType.BINARY)) {
+                // add contentType parameter
+                if (haveDifferentContentTypes(requests) && !hasContentTypeParameter(request)) {
+                    Parameter contentTypeParameter = createContentTypeParameter(request, operation);
+                    request.getParameters().add(getBinarySchemaBodyOrContentLengthParameterIndex(request.getParameters()), contentTypeParameter);
+                    if (contentTypeParameter.isRequired()) {
+                        request.getSignatureParameters().add(getBinarySchemaBodyOrContentLengthParameterIndex(request.getSignatureParameters()), contentTypeParameter);
+                    }
+                }
+                selectedRequest = request;
+                break;
+            }
+        }
+        return selectedRequest;
+    }
+
+    /**
+     *
+     * @param request the input request
+     * @return true if there is parameter in the request named "contentType", otherwise, return false
+     */
+    private static boolean hasContentTypeParameter(Request request) {
+        for (Parameter parameter : request.getParameters()) {
+            if (parameter.getProtocol() != null && parameter.getProtocol().getHttp() != null
+                    && RequestParameterLocation.HEADER.equals(parameter.getProtocol().getHttp().getIn())
+                    && parameter.getLanguage() != null && parameter.getLanguage().getJava() != null
+                    && "Content-Type".equalsIgnoreCase(parameter.getLanguage().getJava().getSerializedName())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     *
+     * @param requests a list of requests
+     * @return true if the requests have different content types, otherwise return false
+     */
+    private static boolean haveDifferentContentTypes(List<Request> requests) {
```

Comment:
Why does this take a list of requests? Should this just look at a single request and return true if that Request takes different content types?

Reply:
Because there can be two requests in one operation, one request's content type is image/jpeg, another request's content type is application/json. We need to verify if the operation contains two content types, so we pass in a list of requests under this operation.

Reply:
It relates to https://github.com/Azure/autorest/blob/main/docs/openapi/request-body.md

m4 be smart enough to make 2 requests sometimes, one for JSON, one for binary.
But since DPG only have BinaryData, we need to combine them to single method.